pagerduty/data_source_pagerduty_event_orchestrations.go

```diff
@@ -0,0 +1,127 @@
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func dataSourcePagerDutyEventOrchestrations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePagerDutyEventOrchestrationsRead,
+
+		Schema: map[string]*schema.Schema{
+			"search": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_orchestrations": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"integration": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"parameters": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"routing_key": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												"type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePagerDutyEventOrchestrationsRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reading PagerDuty Event Orchestrations")
+
+	searchName := d.Get("search").(string)
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		resp, _, err := client.EventOrchestrations.List()
```

Comment:
Please let's separate this `List` call retries from the `Get` retries. We shouldn't be doing nested retries.

Reply:
@imjaroiswebdev I'm not sure to understand as this is the same code than pagerduty/data_source_pagerduty_event_orchestration.go.

Could you give to me an example ?

Reply:
Even though this code is similar to the one of single EO Data Source, what I mean with We shouldn't be nesting retries is that, because the code is triggering a retry inside a loop that is being scoped outside the loop, all the already done calls that probably resulted **ok** would be re executed, and that's why We should in this case scope the retry to each iteration of the loop, to avoid the abuse of the API calls. You can get inspiration from https://github.com/PagerDuty/terraform-provider-pagerduty/blob/5431f120ba880507d6069970f4f50e8391f2b568/pagerduty/resource_pagerduty_event_orchestration_path_unrouted.go#L215 on how to scope the retries inside each loop iteration. @fgouteroux again, please don't hesitate on asking for more help, to be more elaborated or even if you need it, I can directly push a commit with the refactor to solve this particular comment. I'll be happy to do it ✌🏽

Reply:
if you offer to do it, I accept. Thanks